packages/snaps-utils/src/snaps.ts

```diff
@@ -176,31 +179,52 @@ export class ProgrammaticallyFixableSnapError extends Error {
 }
 
 /**
- * Calculates the Base64-encoded SHA-256 digest of a Snap source code string.
+ * Gets a checksummable manifest by removing the shasum property and reserializing the JSON using a deterministic algorithm.
  *
- * @param sourceCode - The UTF-8 string source code of a Snap.
+ * @param manifest - The manifest itself.
+ * @returns A virtual file containing the checksummable manifest.
+ */
+function getChecksummableManifest(
+  manifest: VirtualFile<SnapManifest>,
+): VirtualFile {
+  const manifestCopy = manifest.clone() as any;
+  delete manifestCopy.result.source.shasum;
```

Comment:
Are there any pro/against of deleting the property vs setting it to empty string?

Reply:
I assume that an empty string will still leave the key of the JSON in the stringified version, e.g.
`'{"shasum":""}'` and that seems weird to include in the checksum calculation when we can just omit the key entirely.

Reply:
Both are kinda weird.

I'm just wondering aloud if there any security considerations for either.

Reply:
I can't see why there would be, but I'm not 100% sure.

Reply:
Nothing required for this, just to keep in mind, resolving